File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left),
 * right(right) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        vector<vector<int>> result;
        queue<TreeNode*> store;
        if (root != nullptr) store.push(root);
        while (!store.empty()) {
            vector<int> cur_level;
            int cur_level_size = store.size();
            for (int i = 0; i < cur_level_size; i++) {
                TreeNode* cur_node = store.front();
                cur_level.push_back(cur_node->val);
                store.pop();
                if (cur_node->left)
                    store.push(cur_node->left);
                if (cur_node->right)
                    store.push(cur_node->right);
            }
            result.push_back(cur_level);
        }
        return result;
    }
};

File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def levelOrder(self, root):
        """
        :type root: TreeNode
        :rtype: List[List[int]]
        """
        result = []
        queue = []
        if root:
            queue.append(root)
        while queue:
            cur_level = []
            cur_size = len(queue)
            for i in range(cur_size):
                node = queue.pop(0)                
                cur_level.append(node.val)
                if node.left:
                    queue.append(node.left)
                if node.right:
                    queue.append(node.right)
            result.append(cur_level)
        return result

    def levelOrder_cursive(self, root):
        if not root:
            return []
        result = []
        def traverse(node, level):
            if not node:
                return
            if len(result) == level:
                result.append([])
            result[level].append(node.val)
            traverse(node.left, level+1)
            traverse(node.right, level+1)
        traverse(root, 0)
        return result

        